internal/sink/sink.go

```diff
@@ -209,25 +216,55 @@ func getUrls(urls []string) (int, error) {
 
 func (whs WebhookSink) Send(secret, acceptType string, msg *wrp.Message) error {
 	var errs error
+
+	whs.ch <- msg
+	//add msg to current batch
+	//check if batch is full or linger has been met
+	//no rush to return - have to implement a sleeper
+	//move this if else block into the long running go routine
+
+	return errs
+}
+
+func (whs WebhookSink) listen() {
+	ticker := time.NewTicker(whs.BatchLinger) //TODO: setting to this for now - open to suggestions
+	select {
+	case msg := <-whs.ch:
+		whs.msgs = append(whs.msgs, msg)
+		if len(whs.msgs) == whs.BatchMessages {
+			go whs.batch()
+		}
+	case <-ticker.C:
+		go whs.batch()
+		//TODO: do we need to set a new ticker?
+	}
+}
```

Comment:
we can avoid potential issues if we do the following:
- manage the msgs list locally rather than storing it in the struct
- it's not a good idea to call `batch()` as a goroutine, it'll be easier to have `batch()` return a `ready` channel.
  -  `batch()` should be called when `ready` is closed or nil, you'll have to modify the structure of `batch()` and this function
- instead of `go whs.batch()`, use the goroutine on `v2.send(v2.secret, v2.acceptType, msg)`


We also need a way to stop consuming `<-whs.ch` when `updateUrls`/`Update` are called and resume afterwards 

Other notes:
- rename `listen()` to `batchMsgs()`
- rename `batch()` to `sendBatch()`